tests/end_to_end/enums/mwcc-o4p-out.c

```diff
@@ -0,0 +1,15 @@
+enum my_enum test(enum my_enum x) {
+    if (x != 2) {
+        if (x < 2) {
+            if (x != 0) {
+                return ZERO;
+            }
+            return array->unk0;
+        }
+        if (x != 4) {
+            return ZERO;
+        }
+        return array->unk8;
+    }
+    return array->unk4;
+}
```

Comment:
This test fails to detect the switch statement, and the types are not fully unified across comparisons -- but both of those are separate from this PR.